src/zepben/eas/client/eas_client.py

```diff
@@ -30,45 +30,73 @@ def __init__(
         self,
         host: str,
         port: int,
-        client_id: Optional[str] = None,
+        protocol: str = "https",
         token_fetcher: Optional[ZepbenTokenFetcher] = None,
+        client_id: Optional[str] = None,
         client_secret: Optional[str] = None,
         username: Optional[str] = None,
         password: Optional[str] = None,
-        protocol: str = "https",
-        verify_certificate: bool = False,
+        verify_certificate: bool = True,
         ca_filename: Optional[str] = None,
         session: ClientSession = None,
         json_serialiser = None
     ):
         """
-        :param host: The host string of the Evolve App Server, including the protocol, e.g."https://evolve.local"
+        Construct a client for the Evolve App Server. If the server is HTTPS, authentication may be configured.
+        Authentication may be configured in one of two ways:
+            - Specifying a ZepbenTokenFetcher directly via the token_fetcher parameter
+            - Specifying the client ID of the Auth0 via the client_id parameter, plus one of the following:
+                - The client secret via the client_secret parameter (M2M authentication)
+                - A username and password pair via the username and password parameters (account authentication)
+              If this method id used, the auth configuration will be fetched from the Evolve App Server at the path
+              "/api/config/auth".
+
+        Address parameters:
+        :param host: The domain of the Evolve App Server, including the protocol, e.g. "evolve.local"
         :param port: The port on which to make requests to the Evolve App Server, e.g. 7624
-        :param verify_certificate: Set this to False to disable certificate verification
-        :param ca_filename: Path to CA file to use for verification.
-        :param session: aiohttp CLientSession to use, if not provided a new session will be created for you. You should typically only use one aiohttp session
-        per application.
-        :param json_serialiser: JSON serialiser to use for requests e.g ujson.dumps. Defaults to json.dumps
+        :param protocol: The protocol of the Evolve App Server. Should be either "http" or "https". Must be "https" if
+                         auth is configured. (Defaults to "https")
+
+        Authentication parameters:
+        :param token_fetcher: A ZepbenTokenFetcher used to fetch auth tokens for access to the Evolve App Server.
```

Comment:
Also here move this to below all the username/password options as its not preferred